src/portable/renesas/rusb2/dcd_rusb2.c

```diff
@@ -1022,23 +1002,14 @@ void dcd_int_handler(uint8_t rhport)
   // Buffer ready
   if ( is0 & RUSB2_INTSTS0_BRDY_Msk ) {
     const unsigned m = rusb->BRDYENB;
-    unsigned s = rusb->BRDYSTS & m;
+    const unsigned s = rusb->BRDYSTS & m;
     /* clear active bits (don't write 0 to already cleared bits according to the HW manual) */
     rusb->BRDYSTS = ~s;
-    while (s) {
-#if defined(__CCRX__)
-      static const int Mod37BitPosition[] = {
-        -1, 0, 1, 26, 2, 23, 27, 0, 3, 16, 24, 30, 28, 11, 0, 13, 4,
-        7, 17, 0, 25, 22, 31, 15, 29, 10, 12, 6, 0, 21, 14, 9, 5,
-        20, 8, 19, 18
-      };
 
-      const unsigned num = Mod37BitPosition[(-s & s) % 37];
-#else
-      const unsigned num = __builtin_ctz(s);
-#endif
-      process_pipe_brdy(rhport, num);
-      s &= ~TU_BIT(num);
+    for (unsigned p = 0; p < PIPE_COUNT; ++p) {
```

Comment:
Changing to a `for` statement, I agree that eliminating the need for conditional compilation is beneficial. However, considering execution speed, sticking with the original code that uses bitwise operations might be advantageous. In my estimation, it’s unlikely that multiple bits in BRDYSTS will be set simultaneously. Therefore, the `for` statement would mostly be looping without doing. By using bitwise operations, the loop can be executed only the number of bits that are set in BRDYSTS, avoiding unnecessary iterations. Additionally, ARM CPUs have instructions like CLZ and RBIT that allow efficient bit manipulation in just a few cycles, making this approach particularly effective. 
Although I don’t have an idea, I’d like to explore ways to use bitwise operations without conditional compilation in the future.

Reply:
thank you very much for detailed explanation. This makes sense, I didn't notice this, it is indeed very good piece of code for run time. I have reverted the changes, although forloop is simpler to understand, this is worth keeping since it is in isr.